From: Marco Nenciarini <mnencia@debian.org>
Date: Sat, 5 Mar 2011 11:19:07 +0100
Updated: Wed, 7 Mar 2012 10:43:09 -0500
Subject: split-protocols
Description: Set default protocols value as empty and enable each
 protocol in its own configuration file
---
 doc/example-config/dovecot.conf                    |    4 ++--
 .../doc/example-config/conf.d/20-managesieve.conf  |    3 ---
 src/config/all-settings.c                          |    2 +-
 src/master/master-settings.c                       |    2 +-
 4 files changed, 4 insertions(+), 7 deletions(-)

Index: dovecot/doc/example-config/dovecot.conf
===================================================================
--- dovecot.orig/doc/example-config/dovecot.conf	2013-08-19 11:22:26.322584245 -0400
+++ dovecot/doc/example-config/dovecot.conf	2013-08-19 11:22:26.314584463 -0400
@@ -20,8 +20,8 @@
 # options. The paths listed here are for configure --prefix=/usr
 # --sysconfdir=/etc --localstatedir=/var
 
-# Protocols we want to be serving.
-#protocols = imap pop3 lmtp
+# Enable installed protocols
+!include_try /usr/share/dovecot/protocols.d/*.protocol
 
 # A comma separated list of IPs or hosts where to listen in for connections. 
 # "*" listens in all IPv4 interfaces, "::" listens in all IPv6 interfaces.
Index: dovecot/pigeonhole/doc/example-config/conf.d/20-managesieve.conf
===================================================================
--- dovecot.orig/pigeonhole/doc/example-config/conf.d/20-managesieve.conf	2013-08-19 11:22:26.322584245 -0400
+++ dovecot/pigeonhole/doc/example-config/conf.d/20-managesieve.conf	2013-08-19 11:22:26.314584463 -0400
@@ -2,9 +2,6 @@
 ## ManageSieve specific settings
 ##
 
-# Uncomment to enable managesieve protocol:
-#protocols = $protocols sieve
-
 # Service definitions
 
 #service managesieve-login {
Index: dovecot/src/config/all-settings.c
===================================================================
--- dovecot.orig/src/config/all-settings.c	2013-08-19 11:22:26.322584245 -0400
+++ dovecot/src/config/all-settings.c	2013-08-19 11:22:26.318584354 -0400
@@ -2431,7 +2431,7 @@
 	.libexec_dir = PKG_LIBEXECDIR,
 	.instance_name = PACKAGE,
 	.import_environment = "TZ" ENV_SYSTEMD ENV_GDB,
-	.protocols = "imap pop3 lmtp",
+	.protocols = "",
 	.listen = "*, ::",
 	.ssl = "yes:no:required",
 	.default_internal_user = "dovecot",
Index: dovecot/src/master/master-settings.c
===================================================================
--- dovecot.orig/src/master/master-settings.c	2013-08-19 11:22:26.322584245 -0400
+++ dovecot/src/master/master-settings.c	2013-08-19 11:22:26.318584354 -0400
@@ -216,7 +216,7 @@
 	.libexec_dir = PKG_LIBEXECDIR,
 	.instance_name = PACKAGE,
 	.import_environment = "TZ" ENV_SYSTEMD ENV_GDB,
-	.protocols = "imap pop3 lmtp",
+	.protocols = "",
 	.listen = "*, ::",
 	.ssl = "yes:no:required",
 	.default_internal_user = "dovecot",
